src/functional/otf/compilation/compiler.py

```diff
@@ -0,0 +1,84 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import annotations
+
+import dataclasses
+import pathlib
+from typing import Protocol, TypeVar
+
+from functional.otf import languages, stages, step_types, workflow
+from functional.otf.compilation import build_data, cache, importer
+from functional.otf.step_types import LS, SrcL, TgtL
+
+
+SL = TypeVar("SL", bound=languages.LanguageTag)
+ST = TypeVar("ST", bound=languages.LanguageSettings)
+NT = TypeVar("NT")
```

Comment:
Can you give these some more meaningful names?

Reply:
Do you really want to read those meaningful names in the type hints?